src/Tasks/Common/Resources/Strings.resx

```diff
@@ -952,5 +952,9 @@ You may need to build the project on another operating system or architecture, o
     <value>NETSDK1215: Targeting .NET Standard prior to 2.0 is no longer recommended. See {0} for more details.</value>
     <comment>{StrBegin="NETSDK1215: "}</comment>
   </data>
-  <!-- The latest message added is TargetFrameworkIsNotRecommended. Please update this value with each PR to catch parallel PRs both adding a new message -->
+  <data name="MicrosoftNetSdkCompilersToolsetNotFound" xml:space="preserve">
+    <value>NETSDK1216: Package Microsoft.Net.Sdk.Compilers.Toolset is not downloaded. Either run NuGet package restore or set property 'BuildWithNetFrameworkHostedCompiler' to 'false' to force use of the compiler bundled with MSBuild.</value>
```

Comment:
I want to change the wording of this a bit but am having trouble getting it just right. The current message says what is wrong (the package wasn't downloaded) and how to fix it (restore or toggle the flag), but it doesn't say _why_ the problem is a problem. I want something like

> NETSDK1216: Package Microsoft.Net.Sdk.Compilers.Toolset is was not downloaded but was inferred to be required based on the current build environment. Either run NuGet package restore or set property 'BuildWithNetFrameworkHostedCompiler' to 'false' to force use of the compiler bundled with MSBuild.

But that's not much better :-/

Reply:
> property 'BuildWithNetFrameworkHostedCompiler' to 'false' to force use of the compiler bundled with MSBuild

I'm concerned about this recommendation. Once this error is hit the product is essentially in a broken state (cannot find vital components of the product). Flipping in this property is essentially silencing that signal that something is wrong. It then leads customers into the bad behaviors we saw before.

Reply:
> Once this error is hit the product is essentially in a broken state (cannot find vital components of the product)

I'm not sure that's correct, you can simply be using some internal company feed instead of the public nuget.org. In standard console apps or whatever that used to be completely fine.

> Flipping in this property is essentially silencing that signal that something is wrong. It then leads customers into the bad behaviors we saw before.

Sure but don't we want to tell customers how to suppress this error so they can unblock themselves - given the error is effectively a breaking change?

Reply:
Perhaps something like:

> NETSDK1216: Package Microsoft.Net.Sdk.Compilers.Toolset is not downloaded but it is needed because your MSBuild and SDK versions are mismatched. Ensure it is available in your NuGet feeds and then run NuGet package restore from Visual Studio or MSBuild.

Customers wanting a workaround can find it in the package's docs (https://github.com/dotnet/sdk/pull/42212).

I've also tried to emphasize that you should restore via msbuild because `dotnet restore` wouldn't work.
Maybe `dotnet restore /p:BuildWithNetFrameworkHostedCompiler=true` would work and we can add that to the package docs.